contentcuration/contentcuration/frontend/shared/views/StudioImmersiveModal.vue

```diff
@@ -0,0 +1,139 @@
+<template>
+
+  <div
+    v-if="value"
+    class="modal-wrapper"
+  >
+    <KToolbar
+      :textColor="dark ? 'white' : 'black'"
+      :style="{ backgroundColor: toolbarBackgroundColor }"
+    >
+      <template #icon>
+        <KIconButton
+          icon="close"
+          :ariaLabel="$tr('close')"
+          :tooltip="$tr('close')"
+          :color="dark ? $themeTokens.textInverted : $themeTokens.text"
+          data-test="close"
+          @click="$emit('input', false)"
+        />
+      </template>
+
+      <template #default>
+        <span class="notranslate toolbar-title">
+          <slot name="header">{{ title }}</slot>
+        </span>
+      </template>
+
+      <template #actions>
+        <slot name="action"></slot>
+      </template>
+    </KToolbar>
+
+    <StudioOfflineAlert :offset="64" />
+
+    <StudioPage
+      :offline="offline"
+      :marginTop="contentMarginTop"
+    >
+      <slot></slot>
+    </StudioPage>
+  </div>
+
+</template>
+
+
+<script>
+
+  import { mapState } from 'vuex';
+  import StudioOfflineAlert from './StudioOfflineAlert';
+  import StudioPage from './StudioPage';
+
+  export default {
+    name: 'StudioImmersiveModal',
+    components: {
+      StudioOfflineAlert,
+      StudioPage,
+    },
+    props: {
+      value: {
+        type: Boolean,
+        default: false,
+      },
+      title: {
+        type: String,
+        required: false,
+        default: '',
+      },
+      color: {
+        type: String,
+        default: 'appBarDark',
+      },
+      dark: {
+        type: Boolean,
+        default: true,
+      },
+    },
+    computed: {
+      ...mapState({
+        offline: state => !state.connection.online,
+      }),
+      toolbarBackgroundColor() {
+        return this.color === 'appBarDark'
+          ? this.$themePalette.grey.v_900
+          : this.$themeTokens[this.color] || this.color;
+      },
+      contentMarginTop() {
+        return this.offline ? 112 : 64;
+      },
+    },
+    mounted() {
+      this.hideHTMLScroll(true);
+      const handleKeyDown = event => {
+        if (event.key === 'Escape') {
+          this.$emit('input', false);
+        }
+      };
+      document.addEventListener('keydown', handleKeyDown);
+      this.$once('hook:beforeDestroy', () => {
+        this.hideHTMLScroll(false);
+        document.removeEventListener('keydown', handleKeyDown);
+      });
+    },
+    methods: {
+      hideHTMLScroll(hidden) {
```

Comment:
If the `html` element had a custom overflow style before the modal opened, it would be lost when the modal closes due to this method.

Can you explore alternative approaches that are more robust in this regard? There are more techniques to achieve this. I imagine we would use something fairly straightforward and CSS class based. Before you re-implement code, you can share a brief summary of options you're thinking about here and then we can decide.

Reply:
Yes, that sounds good to me :+1: